### PR TITLE
Renesas RZN2L support

### PR DESCRIPTION
**Pull request by miyazakh**
# Description
Renesas RZN2L support

# Testing

Run Crypt, Bench, TLS client and server

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [x] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by miyazakh** [comment]
retest this please

File: IDE/Renesas/e2studio/RZN2L/test/src/wolfSSL/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (cconlon) [inline IDE/Renesas/e2studio/RZN2L/test/src/wolfSSL/.gitignore:2]: Do we need this `wolfSSL` directory? Seems like it is empty apart from this `.gitignore`.

File: IDE/Renesas/e2studio/RZN2L/test/src/wolfssl_dummy.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <wolfssl/wolfcrypt/wc_port.h>
+
+#define YEAR 2023
+#define MON  5
```
Comment (cconlon) [inline IDE/Renesas/e2studio/RZN2L/test/src/wolfssl_dummy.c:25]: Should we update this to the current month? (ie: `9`).

File: wolfcrypt/benchmark/benchmark.c [hunk 3]
```diff
@@ -2412,8 +2424,12 @@ static void* benchmarks_do(void* args)
         !defined(NO_HW_BENCH)
         bench_aes_aad_options_wrap(bench_aesgcm, 1);
     #endif
-
-        bench_gmac();
+        #ifndef NO_SW_BENCH
```
Comment (cconlon) [inline wolfcrypt/benchmark/benchmark.c:2427]: Spacing - this line needs to back up one tab

File: wolfcrypt/src/random.c [hunk 4]
```diff
@@ -3476,7 +3476,15 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
     }
 #elif defined(WOLFSSL_RENESAS_FSPSM) || \
           defined(WOLFSSL_RENESAS_FSPSM_CRYPTONLY)
+
+   #if defined(WOLFSSL_RENESAS_SCEPROTECT)
     #include "r_sce.h"
+    #define R_RANDOM_GEN(b) R_SCE_RandomNumberGenerate(b)
```
Comment (cconlon) [inline wolfcrypt/src/random.c:3482]: Please adjust spacing in this section to match coding standards - tabs should be 4 spaces.  Looks like these several lines use 1-space in after #if/elif/etc.

File: wolfcrypt/test/test.c [hunk 5]
```diff
@@ -11304,13 +11329,19 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t aes256_test(void)
         0xf5,0x8c,0x4c,0x04,0xd6,0xe5,0xf1,0xba,
         0x77,0x9e,0xab,0xfb,0x5f,0x7b,0xfb,0xd6
     };
-
+#ifdef HAVE_RENESAS_SYNC
+    byte *key =
+                (byte*)guser_PKCbInfo.wrapped_key_aes256;
+    int kyeSz = (256/8);
```
Comment (cconlon) [inline wolfcrypt/test/test.c:11335]: Should this variable be called `keySz`?  `kyeSz` seems like a typo.

**Review by cconlon — CHANGES_REQUESTED** [review]

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 6]
```diff
@@ -0,0 +1,536 @@
...
+## Setup Steps and Build wolfSSL Library
+
+1.) Import  projects from [File]->[Open projects from File System]
+
++ Select folder at /path/to/wolfssl/IDE/Renesas/e2studio/RZN2L/test
+
+2.) Create a `dummy_application` Renesas RZ/N C/C++ FSP Project.
+
++ Click File->New->`RZ/N C/C++ FSP Project`.
++ Enter project name `dummu_application`.
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:63]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 7]
```diff
@@ -0,0 +1,536 @@
...
+
++ Run peer wolfSSL server
+
++ RSA sign and verify use, launch server with the following option
+```
+$./examples/server/server -b -d -i -v 4
+```
+
++ For ECDSA sign and verify use, 
+Enable the `USE_CRT_BUFFER_256` macro in `wolfssl_demo.h`
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:152]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 8]
```diff
@@ -0,0 +1,536 @@
...
+Disble the `USE_CRT_BUFFER_2048` macro in `wolfssl_demo.h`
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:153]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 9]
```diff
@@ -0,0 +1,536 @@
...
+
++ Run peer wolfSSL server
+
++ RSA sign and verify use, launch server with the following option
+```
+$./examples/server/server -b -d -i -v 3
+```
+
++ For ECDSA sign and verify use, 
+Enable the `USE_CRT_BUFFER_256` macro in `wolfssl_demo.h`
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:214]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 10]
```diff
@@ -0,0 +1,536 @@
...
+
++ Server IP address can be changed by the following line in `rzn2l_tst_thread_entry.c`.
+```
+static const byte ucIPAddress[4]          = { 192, 168, 11, 241 };
+```
+
+3.) Build test_RZN2L project
+
++ For ECDSA sign and verify use, 
+Enable the `USE_CRT_BUFFER_256` macro in `wolfssl_demo.h`
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:281]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/test/src/wolfSSL/.gitignore (hunk 1 above)
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/test/src/wolfSSL/.gitignore:2]: The folder is a place folder for e2studio project. In the folder, there are  source files which are linked to ssl.c, tls.c and so on in wolfssl source tree  when opening the project file.

File: IDE/Renesas/e2studio/RZN2L/test/src/wolfssl_dummy.c (hunk 2 above)
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/test/src/wolfssl_dummy.c:25]: That's true.

File: wolfcrypt/benchmark/benchmark.c (hunk 3 above)
Comment (miyazakh) [reply wolfcrypt/benchmark/benchmark.c:2427]: Thanks, fixed.

File: wolfcrypt/src/random.c (hunk 4 above)
Comment (miyazakh) [reply wolfcrypt/src/random.c:3482]: Thanks, fixed.

File: wolfcrypt/test/test.c (hunk 5 above)
Comment (miyazakh) [reply wolfcrypt/test/test.c:11335]: Thanks, Fixed.

File: IDE/Renesas/e2studio/RZN2L/test/src/wolfCrypt/.gitignore [hunk 11]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/test/src/wolfCrypt/.gitignore:2]: The folder is a place folder for e2studio project. In the folder, there are source files which are linked to aes.c, sha.c and so on in wolfcrypt source tree when opening the project file.

File: IDE/Renesas/e2studio/RZN2L/test/src/serial_io/.gitignore [hunk 12]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/test/src/serial_io/.gitignore:2]: The folder is a place folder for e2studio project. In the folder, there are source files which are copied for uart print.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 13]
```diff
@@ -0,0 +1,536 @@
...
+ECDHE [      SECP256R1]   256    agree         2 ops took 1.465 sec, avg 732.500 ms, 1.365 ops/sec
+ECDSA [      SECP256R1]   256     sign         2 ops took 8.049 sec, avg 4024.500 ms, 0.248 ops/sec
+ECDSA [      SECP256R1]   256   verify         2 ops took 2.794 sec, avg 1396.1000 ms, 0.716 ops/sec
+CURVE  25519  key gen         2 ops took 1.077 sec, avg 538.500 ms, 1.857 ops/sec
+CURVE  25519    agree         2 ops took 1.077 sec, avg 538.500 ms, 1.857 ops/sec
+Benchmark complete
+ End wolfCrypt Benchmark
+```
+**Note**
+`SHA1/224/256/384/512` and `Randome generation` of RSIP driver are enabled at the sampele output above.
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:535]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 14]
```diff
@@ -0,0 +1,536 @@
...
+logging  test passed!
+time test passed!
+mutex    test passed!
+crypto callback test passed!
+Test complete
+ End wolfCrypt Test
+```
+
+**Note**
+`SHA1/224/256/384/512` and `Randome generation` of RSIP driver are enabled at the sampele output above while running wolfCrypt test.
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:475]: Thanks, fixed.

File: IDE/Renesas/e2studio/RZN2L/README.md [hunk 15]
```diff
@@ -0,0 +1,536 @@
...
+
++ Server IP address can be changed by the following line in `rzn2l_tst_thread_entry.c`.
+```
+static const byte ucIPAddress[4]          = { 192, 168, 11, 241 };
+```
+
+3.) Build test_RZN2L project
+
++ For ECDSA sign and verify use, 
+Enable the `USE_CRT_BUFFER_256` macro in `wolfssl_demo.h`
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RZN2L/README.md:361]: Thanks, fixed.

**Review by cconlon — APPROVED** [review]

**Pull request merged**